server/base/src/main/java/org/apache/accumulo/server/util/ManagerMetadataUtil.java

```diff
@@ -204,6 +204,8 @@ public static void replaceDatafiles(ServerContext context, KeyExtent extent,
     tablet.putZooLock(zooLock);
 
     tablet.mutate();
+    // Write candidates twice to avoid a possible race condition when removing InUse candidates
```

Comment:
The previous candidate writing could probably use a comment as well, to 1) link it to here, and 2) to specify how it is ordered relative to the tablet mutation that removes the file entries from the tablet.

Reply:
Added an additional comment. Relative order is referenced in regards to the mutation.